File: core-sdk/models/driver.ts
/** Currency amounts in smallest unit (paise) unless noted */
export type DriverStatus = 'Active' | 'Inactive';

export interface Driver {
  id: string;
  name: string;
  vrn: string;
  status: DriverStatus;
  /** Balance in paise for API parity */
  cardBalancePaise: number;
}

export interface DriverUpdatePayload {
  id: string;
  name?: string;
  status?: DriverStatus;
  /** Balance in paise */
  cardBalancePaise?: number;
}

export interface FleetDriverProfile {
  id: string;
  name: string;
  initials: string;
  mobile: string;
  maskedMobile: string;
  registered: boolean;
}

export type BindingState = 'ACTIVE' | 'PENDING_ACCEPTANCE';

export type AuthMode = 'vehicle_linked' | 'shift_based' | 'trip_linked';

export type ScanPayStatus =
  | 'always_available'
  | 'in_window'
  | 'locked_unpaired'
  | 'locked_repair';

export interface FleetBinding {
  id: string;
  vrn: string;
  fo: string;
  authMode: AuthMode;
  state: BindingState;
  paired: boolean;
  scanPayStatus: ScanPayStatus;
  balance?: number;
  cardBalance?: number;
  incentiveBalance?: number;
  spendLimit?: number;
  shiftDays?: string[];
  shiftStart?: string;
  shiftEnd?: string;
  shiftEndsIn?: string;
  tripDate?: string;
  tripStart?: string;
  tripEnd?: string;
  tripEndsIn?: string;
  origin?: string;
  destination?: string;
  assignedBy?: string;
  validPairingCode?: string;
  repairReason?: string;
}

export interface FleetTransaction {
  id: string;
  station: string;
  vrn: string;
  amount: number;
  date: string;
  type: string;
  quantity?: string;
  status: string;
}

export interface FleetVehicleSummary {
  vrn: string;
  company: string;
  authMode: string;
  balance: number;
  limit: number;
}

File: flutter-sdk/lib/src/models/driver.dart
enum DriverStatus { active, inactive }

class Driver {
  const Driver({
    required this.id,
    required this.name,
    required this.vrn,
    required this.status,
    required this.cardBalancePaise,
  });

  final String id;
  final String name;
  final String vrn;
  final DriverStatus status;
  final int cardBalancePaise;

  factory Driver.fromJson(Map<String, dynamic> j) {
    return Driver(
      id: j['id'] as String,
      name: j['name'] as String,
      vrn: j['vrn'] as String,
      status: (j['status'] as String) == 'Inactive'
          ? DriverStatus.inactive
          : DriverStatus.active,
      cardBalancePaise: (j['cardBalancePaise'] as num).toInt(),
    );
  }

  Driver copyWith({
    String? id,
    String? name,
    String? vrn,
    DriverStatus? status,
    int? cardBalancePaise,
  }) {
    return Driver(
      id: id ?? this.id,
      name: name ?? this.name,
      vrn: vrn ?? this.vrn,
      status: status ?? this.status,
      cardBalancePaise: cardBalancePaise ?? this.cardBalancePaise,
    );
  }
}
